lib/wrapThenable.js

```diff
@@ -27,7 +27,13 @@ function wrapThenable (thenable, reply, store) {
       // the request may be terminated during the reply. in this situation,
       // it require an extra checking of request.aborted to see whether
       // the request is killed by client.
-      if (payload !== undefined || (reply.sent === false && reply.raw.headersSent === false && reply.request.raw.aborted === false)) {
+      if (payload !== undefined || //
+        (reply.sent === false && //
+          reply.raw.headersSent === false &&
+        reply.request.raw.aborted === false &&
+        !reply.request.socket.destroyed
```

Comment:
https://github.com/fastify/fastify/issues/4959#issuecomment-1670355986

`socket` can be `undefined` in some cases.

Reply:
> socket can be undefined in some cases.

do you have an example?

I recognize that sometimes `reply.request.socket.destroyed` returns `undefined`, but the `socket` is always there (in the test suite)

Reply:
> do you have an example?

Hard to provide one, but as I remember it happened when I use autocannon to test.
Here is the comment from core, it may be undefined when calling `destroy`.
https://github.com/nodejs/node/blob/36e89dd13cdf7d5a81e0d4b53c0070dd1790705e/lib/_http_incoming.js#L224-L240